pkg/catalogv2/system/manager_test.go

```diff
@@ -36,3 +50,372 @@ func TestManagerRemovesRelease(t *testing.T) {
+			URLs: []string{"foo"},
+		}
+		rancherRepoV2 = repo.ChartVersion{
+			Metadata: &chart.Metadata{
+				Version: "2.0.0",
+			},
+			URLs: []string{"foo"},
+		}
+	)
+
+	tests := []struct {
+		name            string
+		releases        []*release.Release
+		indexedReleases map[string]repo.ChartVersions
+		desiredCharts   map[desiredKey]map[string]any
+		forceAdopt      bool
+		expectInstalls  map[string]bool
+		expectedErr     error
+	}{
+		{
+			name:     "Updates charts to desired version",
+			releases: []*release.Release{&rancherChartV1, &fleetChartV1},
+			indexedReleases: map[string]repo.ChartVersions{
+				"fleet":           {&fleetRepoV1, &fleetRepoV2},
+				"rancher-webhook": {&rancherRepoV1, &rancherRepoV2},
+			},
+			desiredCharts: map[desiredKey]map[string]any{
+				{
+					namespace:    "cattle-fleet-system",
+					name:         "fleet",
+					minVersion:   "1.0.0",
+					exactVersion: "2.0.0",
+				}: {},
+				{
+					namespace:    "cattle-system",
+					name:         "rancher-webhook",
+					minVersion:   "1.0.0",
+					exactVersion: "2.0.0",
+				}: {},
+			},
+			forceAdopt: false,
+			expectInstalls: map[string]bool{
+				"rancher-webhook": true,
+				"fleet":           true,
+			},
+		},
+		{
+			name:     "Keeps installed release matching desired version",
+			releases: []*release.Release{&fleetChartV2, &rancherChartV1},
+			indexedReleases: map[string]repo.ChartVersions{
+				"fleet":           {&fleetRepoV1, &fleetRepoV2},
+				"rancher-webhook": {&rancherRepoV1, &rancherRepoV2},
+			},
+			desiredCharts: map[desiredKey]map[string]any{
+				{
+					namespace:    "cattle-fleet-system",
+					name:         "fleet",
+					minVersion:   "1.0.0",
+					exactVersion: "2.0.0",
+				}: {},
+				{
+					namespace:    "cattle-system",
+					name:         "rancher-webhook",
+					minVersion:   "1.0.0",
+					exactVersion: "2.0.0",
+				}: {},
+			},
+			forceAdopt: false,
+			expectInstalls: map[string]bool{
+				"fleet":           false,
+				"rancher-webhook": true,
+			},
+		},
+		{
+			name:     "Keeps installed release, more recent than desired version",
+			releases: []*release.Release{&rancherChartV1, &fleetChartV3},
+			indexedReleases: map[string]repo.ChartVersions{
+				"fleet":           {&fleetRepoV1, &fleetRepoV2, &fleetRepoV3},
+				"rancher-webhook": {&rancherRepoV1, &rancherRepoV2},
+			},
+			desiredCharts: map[desiredKey]map[string]any{
+				{
+					namespace:    "cattle-fleet-system",
+					name:         "fleet",
+					minVersion:   "1.0.0",
+					exactVersion: "",
+				}: {},
+				{
+					namespace:    "cattle-system",
+					name:         "rancher-webhook",
+					minVersion:   "1.0.0",
+					exactVersion: "2.0.0",
+				}: {},
+			},
+			forceAdopt: false,
+			expectInstalls: map[string]bool{
+				"fleet":           false,
+				"rancher-webhook": true,
+			},
+		},
+		{
+			// This is not a desired use case, but is specified here to emphasize the difference between min
+			// version and exact version behavior.
```

Comment:
nit: this is a desired use case for webhook when a system is restored to a previous version of rancher

Reply:
Oh good point, comment updated. Thanks!